### PR TITLE
Login form

### PR DESCRIPTION
**Pull request by JamesRexMiller4**
### What does this PR do? 

This PR will create the basic html structure for the loginForm component, with some event handlers to update the local state. It also creates a util file with a validateCredentials function that uses RegEx for validating email addresses. Additionally, there are additions made to the interfaces file and the action creator for the user. Adding the userReducer and tests for everything thus mentioned. 

### Where should the reviewer start? 

Reviewer should begin by reviewing the changes to the interfaces file, particularly the UserLoginReceived as that is used for the addUser action creator. Then the user should review the userReducer file and the tests for the userReducer. Finally, they should check out the skeleton fo the loginForm component and see how the handleChanges are being handled. And then check out the test file for that component as well

### Can this be manually tested? 

Yes this can be manually tested by running npm run test, and reviewer should be able to see all the tests passing. Additionally, they can run npm start and confirm that state is being updated as a user types in the email and password

### CODE REVIEW

File: src/Components/LoginForm/LoginForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    it('handlePasswordChange: should update password state when change events occur', () => {
+      const mockEvent: Object = { target: { value: 'password123' } }
+      const wrapper = mount(<LoginForm />);
+      expect(wrapper.find('input').last().getDOMNode().value).toEqual('');
+      wrapper.find('input').last().simulate('change', mockEvent);
+      expect(wrapper.find('input').last().getDOMNode().value).toEqual('password123');
+    });
+  });
+});
```
Comment (RayRedGoose) [inline src/Components/LoginForm/LoginForm.test.tsx]: Good work on testing!

File: src/Components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      </header>
+      <form>
+        <label htmlFor='email'>Email</label>
+        <input id='email' type='text' name='email' placeholder='name@email.com' value={email} 
+        onChange={(e) => handleEmailChange(e)} onBlur={validateEmail}></input>
+        <label htmlFor='password'>Password</label>
+        <input id='password' type='password' name='password' placeholder='********' value={password} 
+        onChange={(e) => handlePasswordChange(e)}></input>
+        <button type='button' onClick={handleSubmit}>Login</button>
+      </form>
```
Comment (RayRedGoose) [inline src/Components/LoginForm/LoginForm.tsx]: Note for next changes: Can you refactor this onClick events and don't use full function there?

File: src/interfaces.ts [hunk 3]
```diff
@@ -14,8 +15,11 @@ export interface UserSignupPosting {
   password: string,
   firstName: string,
   lastName: string,
+  id?: number
 }
 
+// REDUX STORE INTERFACES
+
```
Comment (RayRedGoose) [inline src/interfaces.ts]: Thank you for this comments!

**Comment by RayRedGoose** [comment]
Good work on it! The same issue with my: deleting empty scss file.

File: src/Components/LoginForm/LoginForm.tsx (hunk 2 above)
Comment (JamesRexMiller4) [reply src/Components/LoginForm/LoginForm.tsx]: Yes, I can refactor and trim on the next round

**Comment by JamesRexMiller4** [comment]
> Good work on it! The same issue with my: deleting empty scss file.

File has been removed

**Pull request merged**